main/streams/xp_socket.c

```diff
@@ -402,7 +420,11 @@ static int php_sockop_set_option(php_stream *stream, int option, int value, void
 
 			switch (xparam->op) {
 				case STREAM_XPORT_OP_LISTEN:
+#ifndef __wasi__
 					xparam->outputs.returncode = (listen(sock->socket, xparam->inputs.backlog) == 0) ?  0: -1;
+#else
+					xparam->outputs.returncode = 0;
```

Comment:
This seems wrong to me. Reporting success of listen even though nothing happens...